lib/include/event-driven/vDrawSkin.h

```diff
@@ -0,0 +1,1159 @@
+                rf2.configure(0,NULL);
+                rf2.setVerbose(true);
+
+                if (rf2.check("taxel2Repr")){
+                    check+=1;
+                    std :: cout << "Calibration data";
+                    yarp::os::Bottle calibration_data = *(rf2.find("taxel2Repr").asList());
+                    for (int i = 0; i < calibration_data.size(); i++){
+                            taxel2Repr.push_back(calibration_data.get(i).asInt());
+                    }
+                    handL.initRepresentativeTaxels(taxel2Repr);
+                }
+
+                rf1.setVerbose(false);
+                rf1.setDefaultContext("skinGui");            //overridden by --context parameter
+                rf1.setDefaultConfigFile(filename1); //overridden by --from parameter
+                rf1.configure(0,NULL);
+                rf1.setVerbose(true);
+
+                if (rf1.check("SENSORS")){
+                    check+=1;
+                    std:: cout<< "\n Sensors data"<<std::endl;
+                    yarp::os::Bottle &ini_data = rf1.findGroup("SENSORS");
+
+                    for (int i = 1; i < (ini_data.size()); i++){
+
+                        yarp::os::Bottle sensorConfig(ini_data.get(i).toString());
+
+                        //std::string type(sensorConfig.get(0).asString()); useless since forearm has always same type and hand has a precise numbering for palm and fingers.
+
+                        int id=sensorConfig.get(1).asInt();
+                        double x=sensorConfig.get(2).asDouble();
+                        double y=sensorConfig.get(3).asDouble();
+                        double orientation=sensorConfig.get(4).asDouble();
+                        double gain=sensorConfig.get(5).asDouble();
+                        int    mirror=sensorConfig.get(6).asInt();
+                        int    layoutNum=sensorConfig.get(7).asInt();
+
+                        handL.get_data(id,x,y,orientation,gain,mirror,layoutNum);
+                    }
+                }
+                if(check==2){
+                    yInfo("Building taxel map from file");
+                    for (auto it =handL.repr2TaxelList.begin();it!=handL.repr2TaxelList.end();it++){
+                        for (auto it2 =  it->second.begin(); it2 !=it->second.end(); it2++){
+                            pos.insert(std::pair<int,std::tuple<double, double>>(*it2,handL.makeMap(*it2)));
+                        }
+                    }
+                    xoffset=handL.xoffset;
+                    yoffset=handL.yoffset;
+                    scaling =handL.scaling;
+                    radius = handL.radius;
+                    noise = handL.noise;
+                    max_value = handL.max_value;
+                }
+                else {
+                    yWarning("Unable to build the taxel map from file");
+                    pos = default_pos;
+                }
+            }
+
+        };
+
```

Comment:
A lot of duplicated code that would be hard to maintain in the future. Also is not clear what this "check" variable is. It seems like it is used to move to the next part of the initialization. Probably could be replaced by return statements if the init function cannot proceed for some reasons

Reply:
agreed, for now it "works" but should be made more understandable later